app/controllers/spree/admin/orders_controller_decorator.rb

```diff
@@ -20,58 +20,10 @@
 
   before_filter :require_distributor_abn, only: :invoice
 
-
   respond_to :html, :json
 
-  # Mostly the original Spree method, tweaked to allow us to ransack with completed_at in a sane way
   def index
-    params[:q] ||= {}
-    params[:q][:completed_at_not_null] ||= '1' if Spree::Config[:show_only_complete_orders_by_default]
-    @show_only_completed = params[:q][:completed_at_not_null].present?
-    params[:q][:s] ||= @show_only_completed ? 'completed_at desc' : 'created_at desc'
-
-    # As date params are deleted if @show_only_completed, store
-    # the original date so we can restore them into the params
-    # after the search
-    created_at_gt = params[:q][:created_at_gt]
-    created_at_lt = params[:q][:created_at_lt]
-
-    params[:q].delete(:inventory_units_shipment_id_null) if params[:q][:inventory_units_shipment_id_null] == "0"
-
-    if !params[:q][:created_at_gt].blank?
-      params[:q][:created_at_gt] = Time.zone.parse(params[:q][:created_at_gt]).beginning_of_day rescue ""
-    end
-
-    if !params[:q][:created_at_lt].blank?
-      params[:q][:created_at_lt] = Time.zone.parse(params[:q][:created_at_lt]).end_of_day rescue ""
-    end
-
-    # Changed this to stop completed_at being overriden when present
-    if @show_only_completed
-      params[:q][:completed_at_gt] = params[:q].delete(:created_at_gt) unless params[:q][:completed_at_gt]
-      params[:q][:completed_at_lt] = params[:q].delete(:created_at_lt) unless params[:q][:completed_at_gt]
-    end
-
-    @orders = orders
-
-    # Restore dates
-    params[:q][:created_at_gt] = created_at_gt
-    params[:q][:created_at_lt] = created_at_lt
-
-    respond_with(@orders) do |format|
-      format.html
-      format.json do
-        render json: {
-          orders: ActiveModel::ArraySerializer.new(@orders, each_serializer: Api::Admin::OrderSerializer),
-          pagination: {
-            results: @orders.total_count,
-            pages: @orders.num_pages.to_i,
-            page: params[:page].to_i,
-            per_page: params[:per_page].to_i
-          }
-        }
-      end
-    end
+    # Moved to api. Overriding the action so we only render the page template.
```

Comment:
Ok, so then we render the template and then we request the orders with an Ajax call to the API?

Reply:
Yes. We just render the template and the Angular in the page makes it's request.

Reply:
Future readers may find this confusing without more context. What moved to the api? What is in the page template? Maybe we could write a bit more to explain:
> This action renders the page template without orders. The template contains Javascript calls to fetch the order data from the API (`Api::OrderController`).